Mcu/l431/Src/WS2812.c

```diff
@@ -0,0 +1,61 @@
+#include "WS2812.h"
+
+#include "targets.h"
+
+#ifdef USE_LED_STRIP
+
+#ifndef WS2812_PORT
+    #error "WS2812_PORT is not defined"
+#endif
+
+#ifndef WS2812_PIN
+    #error "WS2812_PIN is not defined"
+#endif
+
+void enableDWT_CycleCounter(void)
+{
+    // Enable TRCENA in the DEMCR (Debug Exception and Monitor Control Register)
+    CoreDebug->DEMCR |= CoreDebug_DEMCR_TRCENA_Msk;
+    // Enable the cycle counter
+    DWT->CTRL |= DWT_CTRL_CYCCNTENA_Msk;
+    // Reset the cycle counter
+    DWT->CYCCNT = 0;
+}
+
+void waitClockCycles(uint16_t cycles)
+{
+    DWT->CYCCNT = 0; // Reset the cycle counter
+    while (DWT->CYCCNT < cycles) {
+        // Wait until the specified number of cycles has passed
+    }
+}
+
+void sendBit(uint8_t inbit)
```

Comment:
static, and a bit less obscure to ask for 40us or 80us (did I get that right?), rather than the CPU MHz calculation

Reply:
Thanks for your review, it need timing control in range from 220ns to 1000ns, I try to create a "wait_ns" but it's not precise as expected, to avoid float number, so this is a approximately timing with cycles to set the T0H, T0L, T1H, T1L

Reply:
ok, thanks